lib/pure/httpclient.nim

```diff
@@ -260,7 +260,7 @@ proc contentLength*(response: Response | AsyncResponse): int =
   ## A `ValueError` exception will be raised if the value is not an integer.
   var contentLengthHeader = response.headers.getOrDefault("Content-Length")
   result = contentLengthHeader.parseInt()
-  doAssert(result >= 0 and result <= high(int32))
+  doAssert result >= 0
```

Comment:
`-1` will crash this as well - if anything, it could raise a `ValueError` - `parseInt` already does that sometimes

Reply:
Suppose it would but imo >=0 zero is something that should never happen. I can't see why a server would send a negative content-size unless it was misconfigured. Although maybe that's something that does happen and I'm just ignorant to it.

Reply:
As noted elsewhere: an assert enables the server to crash the client - this is remote unvalidated data, it should never ever lead to a crash in the client, but rather an exception that the application can handle (asserts can not be caught generally)

Reply:
I see.